js/PulsarFormComponent.js

```diff
@@ -84,9 +84,24 @@ PulsarFormComponent.prototype.initTimePickers = function () {
  */
 PulsarFormComponent.prototype.initDatePickers = function () {
     // Attach basic pikaday to datepicker fields
-    this.$html
-        .find('[data-datepicker=true]')
-        .pikaday({ format: 'DD/MM/YYYY' });
+
+    const datepickers = this.$html.find('[data-datepicker="true"]');
+
+    datepickers.each((index, element) => {
+        const $parent = $(element).parent();
+
+        $parent
+            .find('[data-format=default]')
+            .pikaday({ format: 'DD/MM/YYYY' });
+
+        $parent
+            .find('[data-format=US]')
+            .pikaday({ format: 'MM/DD/YYYY' });
+
+        $parent
+            .find('[data-format=reverse]')
+            .pikaday({ format: 'YYYY/MM/DD' });
+    });
```

Comment:
I'm still not certain all of these `find(...)` calls are necessary.

You will be able to get the `data-format` attribute within this `datepickers.each(...)` with `$(element).attr('data-format')`. I'm unclear as to why we need to find the parent and then query it, this approach would make more sense if the `data-format` and `data-datepicker` existed on separate elements.

Reply:
This change will break any existing datepickers which don't have a format option defined, there should be a default fallback to UK dates when no format is supplied.

This should be addressed while refactoring out the multiple find calls.

Reply:
If you add this to `line 91`

```console.log($(element).attr('data-format'));```

You should see what I'm getting at, you have access to the format without having to query parents and descendants.

Reply:
@nvasileiadis That unfortunately presumes developers will only use the helper, and doesn't cater for hardcoded markup